crypto/data_formats_test.go

```diff
@@ -16,14 +16,93 @@ package crypto
 
 import (
 	"bytes"
+	"reflect"
 	"testing"
 
 	"github.com/google/trillian"
 	"github.com/google/trillian/testonly"
+	"github.com/kr/pretty"
 )
 
 var dh = testonly.MustHexDecode
 
+func TestLogRoot(t *testing.T) {
+	for _, logRoot := range []*LogRootV1{
+		{
+			Version:  1,
```

Comment:
Need test cases exercising version overwriting during serialization.

Reply:
Refactored to remove this sharp edge.

Reply:
Good idea.